daqLogger/src/main.cpp

```diff
@@ -0,0 +1,191 @@
+#include "virtualTimer.h"
+#include "can_interface.h"
+#include "virtualTimer.h"
+
+
+RTC_PCF8523 rtc;
+
+// FAILURE MODES:
+//  Is data all changing? If not, sensor could be broken
+//  Is data not being sent from a specific sensor? If not, could be broken
+
+
+
+// Goals:
+//   Read HP CAN
+//     Time, Motor temp, Inverter temp, Coolant temp, Remaining battery, 
+//     Battery temp, Power output, Throttle, HV Battery voltage
+//   Read LP CAN
+//     Time (RTC), Wheel Speed (one for each wheel), Break temp (for each break),
+//     Break pressure (for front/back breaks), LV Battery voltage
+//   Store information to SD Card
+//     Use CSV
+//     Log every 10 ms:
+//       RTC, Accelerometer
+//     Log every 100 ms:
+//       Wheel Speed, GPS, Suspension
+//     Log every 1000 ms:
+//       Ambient temp, Brake temp, LV Battery voltage, HV Battery voltage
+
+// Initialize canbus based on teensy board
+#if defined(ARDUINO_TEENSY40) || defined(ARDUINO_TEENSY41)
+#include "teensy_can.h"
+// The bus number is a template argument for Teensy: TeensyCAN<bus_num>
+TeensyCAN<1> can_bus{};
+#endif
+
+#ifdef ARDUINO_ARCH_ESP32
+#include "esp_can.h"
+// The tx and rx pins are constructor arguments to ESPCan, which default to TX = 5, RX = 4
+ESPCAN can_bus{GPIO_NUM_17, GPIO_NUM_16};
+#endif
+
+const int kFLCANFrameAddress = 0x400;
+const int kFRCANFrameAddress = 0x401;
+const int kBLCANFrameAddress = 0x402;
+const int kBRCANFrameAddress = 0x403;
+
+CANSignal<float, 0, 16, CANTemplateConvertFloat(0.1), CANTemplateConvertFloat(0), false> wheel_speed_signal{}; 
+CANSignal<float, 16, 16, CANTemplateConvertFloat(0.1), CANTemplateConvertFloat(-40), false> brake_temp_signal{};
+CANRXMessage<2> rx_message_wheel{can_bus, kFLCANFrameAddress, wheel_speed_signal, brake_temp_signal};
+
+const int kGPS_CAN = 0x430;
+const int kACCEL_CAN = 0x431;
+const int kGYRO_CAN = 0x432;
+const int kGYRO_2_CAN = 0x432;
+
+CANSignal<float, 0, 16, CANTemplateConvertFloat(0.1), CANTemplateConvertFloat(0), false> accel_x{}; 
+CANSignal<float, 16, 16, CANTemplateConvertFloat(0.1), CANTemplateConvertFloat(-40), false> accel_y{}; 
+CANSignal<float, 32, 16, CANTemplateConvertFloat(0.1), CANTemplateConvertFloat(-40), false> accel_z{};
+CANRXMessage<3> rx_message_accel{can_bus, kACCEL_CAN, accel_x, accel_y, accel_z};
+
+CANSignal<float, 0, 16, CANTemplateConvertFloat(0.1), CANTemplateConvertFloat(-40), false> gyro_x{}; 
+CANSignal<float, 16, 16, CANTemplateConvertFloat(0.1), CANTemplateConvertFloat(-40), false> gyro_y{}; 
+CANSignal<float, 32, 16, CANTemplateConvertFloat(0.1), CANTemplateConvertFloat(-40), false> gyro_z{}; 
+CANRXMessage<3> rx_message_gyro{can_bus, kGYRO_CAN, gyro_x, gyro_y, gyro_z};
+
+CANSignal<float, 0, 16, CANTemplateConvertFloat(0.1), CANTemplateConvertFloat(-40), true> lon_signal{}; 
+CANSignal<float, 16, 16, CANTemplateConvertFloat(0.1), CANTemplateConvertFloat(-40), true> lat_signal{}; 
+CANRXMessage<2> rx_message_pos{can_bus, kGPS_CAN, lon_signal, lat_signal};
```

Comment:
check DBC files for updated scalars and offsets for rx_message_accel, rx_message_gyro, and rx_message_pos signals